linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/H2Config.scala

```diff
@@ -200,16 +202,30 @@ trait H2SvcConfig extends SvcConfig {
   var _h2Classifier: Option[H2ClassifierConfig] = None
 
   @JsonIgnore
-  def h2BaseResponseClassifier = H2StreamClassifiers.Default
+  def h2Classifier: Option[H2Classifier] =
+    _h2Classifier.map(_.mk)
 
-  @JsonIgnore
-  def h2ResponseClassifier: Option[H2StreamClassifier] =
-    _h2ResponseClassifier
-      .map { c => H2StreamClassifiers.NonRetryableStream(c.mk).orElse(h2BaseResponseClassifier) }
+  @JsonDeserialize(contentAs = classOf[java.lang.Long])
+  var classificationTimeoutMs: Option[Long] = None
+
+  var retryBufferSize: Option[RetryBufferSize] = None
 
   @JsonIgnore
-  override def params(vars: Map[String, String]): Stack.Params = super.params(vars)
-    .maybeWith(h2ResponseClassifier.map(H2StreamClassifier(_)))
+  override def params(vars: Map[String, String]): Stack.Params =
+    super.params(vars)
+      .maybeWith(h2Classifier.map(h2Param.H2Classifier(_)))
+      .maybeWith(classificationTimeoutMs.map { t => ClassificationTimeout(t.millis) })
```

Comment:
this should have a default -- ALSO we should ensure that it fits in the timer granularity ( i think has to be greater than 10ms?)